tr2_keyboard_teleop/src/tr2_keyboard_teleop.py

```diff
@@ -3,95 +3,118 @@
 import curses
 import time
 import rospy
-import sys
-import signal
-import numpy as np
-import math
-import datetime
 from tr2py.tr2 import TR2
 
 tr2 = TR2()
 tr2.setMode(tr2.mode_rotate)
 tr2.release()
 
-jointSelected = 0;
-joints = ["b0", "a0", "a1", "a2", "a3", "a4", "g0", "h0", "h1"]
+JOINTS = [
+    tr2.b0,
+    tr2.a0,
+    tr2.a1,
+    tr2.a2,
+    tr2.a3,
+    tr2.a4,
+    tr2.g0,
+    tr2.h0,
+    tr2.h1
+]
+
+MODES = [
+    ("ROTATE", tr2.mode_rotate),
+    ("BACKDRIVE", tr2.mode_backdrive),
+    ("SERVO", tr2.mode_servo)
+]
+
+
+def get_circular_index_increment(new_index, max_index):
+    if new_index > max_index:
+        # If we tried to go past the end, set back to the beginning
+        new_index = 0
+    elif new_index < 0:
+        # While python will allow negative indices, let's just reset
+        # back to the end of the list to avoid confusion
+        new_index = max_index
+    return new_index
 
-modeSelected = 0
-modes = ["ROTATE","BACKDRIVE","SERVO"]
-
-stop = "FALSE"
 
 def program():
-	global jointSelected, joints, modeSelected, modes, stop, tr2
```

Comment:
All of these variables are accessible. `global` is only necessary if we plan on modifying the referenced variable within the function scope.